cmake/templates/BoostConfig.cmake.in

```diff
@@ -51,10 +50,16 @@ if(NOT TARGET "Boost::boost")
       INTERFACE_INCLUDE_DIRECTORIES "${_root_path}/include"
   )
   if(MSVC)
-    set_target_properties(
+    #stop MSVC from trying to auto-link libs
+	set(compile_definitions "BOOST_ALL_NO_LIB=1")
+    if(@BOOST_ALL_DYN_LINK@) #required for MSVC when using shared libs
+	  set(compile_definitions "${compile_definitions} BOOST_ALL_DYN_LINK")
```

Comment:
> Setting INTERFACE_COMPILE_DEFINITIONS using a target property will not take a list (or rather it will only take the first item)

It is. Try this example:

```cmake
cmake_minimum_required(VERSION 2.8)
project(foo)

add_library(boo::boo INTERFACE IMPORTED)
set_target_properties(
    boo::boo
    PROPERTIES
    INTERFACE_COMPILE_DEFINITIONS "A=1 B=2"
)

add_executable(foo foo.cpp)
target_link_libraries(foo PUBLIC boo::boo)
```

Here is the compile arguments:

```
/usr/bin/g++ -DA="1 B=2" -o ... -c foo.cpp
```

It means that we define macro `A` with value `1 B=2`. I guess it's not what we expect.

> scratch that target_compile_definitions does not work for imported targets

Yes, it works only for the real targets. Imported targets is usually not what user have to create himself, we are hacking the internal level now.